Mithril update 2025-06-04 (#605)
IntersectMBO/cardano-updates: blog/2025-06-04-mithril.md

---
title: Mithril Team Update
slug: 2025-06-04-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> This week, the Mithril team finalized the support for Cardano node v.10.4.1 and UTXO-HD in the aggregator, signer, and client.
> 
> They completed the implementation of the local message submission and local message notification node-to-client mini-protocols in the Pallas library for the DMQ node. They also kept working on extending the aggregator metrics to record the usage of different client types. Additionally, they worked on stabilizing the Cardano DB v2 client commands in the CLI and API in the library, ensuring it is ready for production use.
> 
> Finally, the team started to work on switching the protocol messages encoding to binary, and they continued refactoring the STM cryptographic library.

Reviewer: ```suggestion
This week, the Mithril team delivered support for Cardano node v.10.4.1 and UTXO-HD in the aggregator, signer, and client.

They completed the implementation of the local message submission and local message notification node-to-client mini-protocols in the Pallas library for the DMQ node. They also continued extending the aggregator metrics to record the usage of different client types. Additionally, they worked on stabilizing the Cardano DB v2 client commands in the CLI and API in the library, ensuring readiness for production use.

Finally, the team began switching the protocol message encoding to binary and continued refactoring the STM cryptographic library.
```
